projects/dendrites/permutedMNIST/centroid_for_figure_1b.py

```diff
@@ -0,0 +1,286 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
```

Comment:
Why have a whole complete source file for one figure? Can't we just run existing code with specific parameters? Or is there something special about this version?

Reply:
We use the `PlotHiddenActivations` mixin to create that figure, but the mixin relies on target values. The problem here is that if `share_labels = True`, then the labels for examples in different tasks overlap, so we couldn't generate the figure. I only modification I made here is to have a separate function for evaluation that doesn't squish all labels into {0, 1, ..., 9}, so the plots can be created properly, which is called `tracking_eval`.

Reply:
I agree that it seems wasteful to have a whole script just for this. I can modify the original script so that it has a "2nd evaluation loop" specifically for plotting. What do you think?

Reply:
Ok, sounds good.

Reply:
I removed the script, and added a mixin that runs the additional eval loop using **task labels** instead of **shared target labels** (which is exactly what we want for plotting).

Reply:
👍